src/segment/filter/mod.rs

```diff
@@ -38,3 +45,56 @@ impl BloomConstructionPolicy {
         }
     }
 }
+
+enum FilterType {
+    StandardBloom = 0,
+    BlockedBloom = 1,
+}
+
+impl TryFrom<u8> for FilterType {
+    type Error = ();
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::StandardBloom),
+            1 => Ok(Self::BlockedBloom),
+            _ => Err(()),
+        }
+    }
+}
+
+pub trait AMQFilter: Sync + Send {
+    fn bytes(&self) -> &[u8];
+    fn len(&self) -> usize;
+    fn contains(&self, item: &[u8]) -> bool;
+    fn contains_hash(&self, hash: (u64, u64)) -> bool;
+}
+
+pub struct AMQFilterBuilder {}
+
+impl AMQFilterBuilder {
+    pub fn decode_from<R: Read>(reader: &mut R) -> Result<Box<dyn AMQFilter + Sync>, DecodeError> {
```

Comment:
I couldn't use impl Decode, because we're not returning a AMQFilterBuilder here.

Otherwise, the method signature is the same